src/core/function.js

```diff
@@ -364,7 +364,9 @@ var PDFFunction = (function PDFFunctionClosure() {
         var rmin = encode[2 * i];
         var rmax = encode[2 * i + 1];
 
-        tmpBuf[0] = rmin + (v - dmin) * (rmax - rmin) / (dmax - dmin);
+        // Ensure that this is always a valid number, since rounding errors
+        // may cause it to become NaN (fixes issue6113.pdf).
+        tmpBuf[0] = (rmin + (v - dmin) * (rmax - rmin) / (dmax - dmin)) || 0;
```

Comment:
Really good point, thanks! I'll update the patch to clamp the value to the range of the function.